### PR TITLE
fix(cdk/active-descendant-key-manager): emit change event after setting active styles

### PR DESCRIPTION
**Pull request by Ritboy**
Before this change, the "change" event was triggered before the active styles have been set and it was impossible to subscribe to this event and handle items with updated styles (e.g. call change detection). This change fixes it.

### CODE REVIEW

**Comment by Ritboy** [comment]
@googlebot I fixed it.

File: src/cdk/a11y/key-manager/activedescendant-key-manager.ts [hunk 1]
```diff
@@ -40,13 +40,18 @@ export class ActiveDescendantKeyManager<T> extends ListKeyManager<Highlightable
...
       this.activeItem.setInactiveStyles();
     }
-    super.setActiveItem(index);
+    super.updateActiveItem(index);
     if (this.activeItem) {
       this.activeItem.setActiveStyles();
     }
+    if (previousIndex !== this.activeItemIndex) {
+      this.change.next(this.activeItemIndex);
+    }
```
Comment (jelbourn) [inline src/cdk/a11y/key-manager/activedescendant-key-manager.ts]: This part of the change is probably okay, but changing the public API is technically a breaking change. We'd also need a unit test for this change in behavior.

File: src/cdk/a11y/key-manager/list-key-manager.ts [hunk 2]
```diff
@@ -265,7 +265,7 @@ export class ListKeyManager<T extends ListKeyManagerOption> {
   }
 
   /** Index of the currently active item. */
-  get activeItemIndex(): number | null {
+  get activeItemIndex(): number {
```
Comment (crisbeto) [inline src/cdk/a11y/key-manager/list-key-manager.ts]: You can avoid the breaking change here by reverting the change. If it's something that we should clean up, you can add `@breaking-change 10.0.0 remove `null` from return type` and we'll do it next time we're making breaking changes.

File: src/cdk/a11y/key-manager/activedescendant-key-manager.ts (hunk 1 above)
Comment (crisbeto) [reply src/cdk/a11y/key-manager/activedescendant-key-manager.ts]: This makes it inconsistent with the other key managers (e.g. `FocusKeyManager`) which still have the old event timing. Maybe we should tackle the issue at the base class level?

**Comment by mmalerba** [comment]
Closing due to inactivity

**Pull request closed without merging**